octobot/strategy_optimizer/optimizer_settings.py

```diff
@@ -0,0 +1,170 @@
+                fitness_parameter.FitnessParameter.WEIGHT_KEY: 0,
+                fitness_parameter.FitnessParameter.IS_RATIO_FROM_MAX_KEY: False,
+            },
+        ]
+
+    def parse_optimizer_filter(self, filters):
+        return [
+            optimizer_filter.OptimizerFilter.from_dict(element)
+            for element in filters
+        ]
+
+    def get_default_optimizer_filters(self):
+        return [
+            {
+                optimizer_filter.OptimizerFilter.LEFT_OPERAND_KEY_KEY: commons_enums.BacktestingMetadata.TRADES.value,
+                optimizer_filter.OptimizerFilter.RIGHT_OPERAND_KEY_KEY: None,
+                optimizer_filter.OptimizerFilter.LEFT_OPERAND_VALUE_KEY: None,
+                optimizer_filter.OptimizerFilter.RIGHT_OPERAND_VALUE_KEY: 1,
+                optimizer_filter.OptimizerFilter.OPERATOR_KEY: commons_enums.LogicalOperators.LOWER_THAN.value,
+            },
+            {
+                optimizer_filter.OptimizerFilter.LEFT_OPERAND_KEY_KEY:
+                    commons_enums.BacktestingMetadata.COEFFICIENT_OF_DETERMINATION_MAX_BALANCE.value,
+                optimizer_filter.OptimizerFilter.RIGHT_OPERAND_KEY_KEY: None,
+                optimizer_filter.OptimizerFilter.LEFT_OPERAND_VALUE_KEY: None,
+                optimizer_filter.OptimizerFilter.RIGHT_OPERAND_VALUE_KEY: 0,
+                optimizer_filter.OptimizerFilter.OPERATOR_KEY: commons_enums.LogicalOperators.LOWER_THAN.value,
+            },
+            {
+                optimizer_filter.OptimizerFilter.LEFT_OPERAND_KEY_KEY:
+                    commons_enums.BacktestingMetadata.PERCENT_GAINS.value,
+                optimizer_filter.OptimizerFilter.RIGHT_OPERAND_KEY_KEY: None,
+                optimizer_filter.OptimizerFilter.LEFT_OPERAND_VALUE_KEY: None,
+                optimizer_filter.OptimizerFilter.RIGHT_OPERAND_VALUE_KEY: 0,
+                optimizer_filter.OptimizerFilter.OPERATOR_KEY: commons_enums.LogicalOperators.LOWER_THAN.value,
+            },
+        ]
+
+    def parse_optimizer_constraint(self, constaints):
+        return {
+            element.name: element
+            for element in (
+                optimizer_constraint.OptimizerConstraint.from_dict(element)
+                for element in constaints
+            )
+        }
+
+    def get_default_optimizer_constraints(self):
+        # return [
+        #     {
+        #         optimizer_constraint.OptimizerConstraint.NAME_KEY: "tentacle-user_input",
+        #         optimizer_constraint.OptimizerConstraint.MIN_VAL_KEY: 1,
+        #         optimizer_constraint.OptimizerConstraint.MAX_VAL_KEY: 3,
+        #         optimizer_constraint.OptimizerConstraint.MIN_STEP_KEY: 0.25,
+        #         optimizer_constraint.OptimizerConstraint.MAX_STEP_KEY: 1,
+        #         optimizer_constraint.OptimizerConstraint.STAY_WITHIN_BOUNDARIES_KEY: False,
+        #     }
+        # ]
```

Comment:
What should we do with this code ?

Reply:
it's an example of a constraint declaration